### PR TITLE
Add a broadcaster capability

### PR DESCRIPTION
**Pull request by dealancer**
This PR should be a fix (or at least a step forward fixing) #11.

### CODE REVIEW

**Comment by martin-helmich** [comment]
Thanks for the PR! :+1:

Before looking at the code: Can we maybe give this feature another name? I'd prefer to avoid any possible confusion with the "official" Varnish product -- both for "least astonishment" and also (IANAL, but better safe than sorry) for legal reasons.

Creative suggestions are welcome. 😉

**Comment by dealancer** [comment]
@martin-helmich can we call it just Broadcaster, or do you want to name it something even more different?

**Comment by dealancer** [comment]
@martin-helmich here are some ideas:
* duster, cleaner - because we need to dust off the varnished surface
* distributor, signaller, herald - because we need to deliver a message to multiple Varnish nodes

**Comment by dealancer** [comment]
@martin-helmich what do you think?

File: README.md [hunk 1]
```diff
@@ -15,13 +15,38 @@ This controller is not intended to be a replacement of a regular [ingress contro
...
+```
+
+It supports broadcasting PURGE and BAN requests to all Varnish nodes.
+
+
 The Varnish controller needs the following prerequisites to run:
 
 - A [Go-template](https://golang.org/pkg/text/template/) that will be used to generate a [VCL](https://varnish-cache.org/docs/trunk/users-guide/vcl.html) configuration file
-- A [Kubernetes service](https://kubernetes.io/docs/concepts/services-networking/service/) that will be used as backend for the Varnish controller
+- A [Kubernetes service](https://kubernetes.io/docs/concepts/services-networking/service/) that will be used to determine Varnish (frontend) and application (backend) pods
```
Comment (martin-helmich) [inline README.md]: The wording might be a bit confusing, here. In fact, you'll need _two_ Kubernetes `Service`s -- one for the frontend and one for the backend pods.

File: broadcaster/run.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		b.errors <- err
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	glog.V(5).Infof("received a broadcast request: %+v", r)
+
+	b.mutex.RLock()
```
Comment (martin-helmich) [inline broadcaster/run.go]: Optional nitpick: Maybe just `defer b.mutex.RUnlock()` right here?

**Comment by martin-helmich** [comment]
Sorry, I kind of lost track of this over the past few days. I'd probably prefer something like "Signaller" or maybe "Dispatcher".

I've also had a look at the code -- looking good so far; thanks for all your work! :+1: Apart from the naming thing, I've only had a few minor nitpicks.

**Comment by dealancer** [comment]
@martin-helmich thank you for the review, good catches.

I have renamed Broadcaster to Signaller, updated docs, and used defer for mutex unlocks.

**Comment by dealancer** [comment]
@martin-helmich hi, how are the things going? Hope you are ok. Did you have any chance looking into the updated PR?

**Comment by martin-helmich** [comment]
Hey @dealancer, sorry for the delay. I've been taking a break from work for the past one-and-a-half weeks -- I'll get to it now.

**Comment by martin-helmich** [comment]
Looking good! Many thanks for this major contribution! :+1: :tada:

**Comment by dealancer** [comment]
@martin-helmich great, thank you!

**Pull request merged**